ktor-client/ktor-client-java/jvm/test/io/ktor/client/engine/java/JavaEngineTests.kt

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2014-2020 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.client.engine.java
+
+import io.ktor.client.*
+import io.ktor.client.request.*
+import kotlinx.coroutines.*
+import kotlin.test.*
+
+class JavaEngineTests {
+    @Test
+    fun testClose() {
+        val engine = JavaHttpEngine(JavaHttpConfig())
+        engine.close()
+
+        assertTrue("Java HTTP dispatcher is not working.") {
+            engine.executor.isShutdown
+        }
+    }
+
+    @Test
+    fun testThreadLeak() = runBlocking {
```

Comment:
this test looks flaky

Reply:
Is it failing on your local machine or in CI build? The tricky part in it - to wait when SelectorManager thread exists, by default it uses polling each 3 seconds and the minimal polling interval is 1 second.

Reply:
https://github.com/AdoptOpenJDK/openjdk-jdk11/blob/master/src/java.net.http/share/classes/jdk/internal/net/http/HttpClientImpl.java#L827